packages/uploadthing/src/internal/incompat-node-guard.ts

```diff
@@ -0,0 +1,27 @@
+export function incompatibleNodeGuard() {
+  if (typeof process === "undefined") return;
+
+  let major: number | undefined;
+
+  const maybeNodeVersion = process.versions?.node?.split(".");
+  if (maybeNodeVersion) {
+    [major] = maybeNodeVersion.map((v) => parseInt(v, 10));
+  }
+
+  const maybeNodePath = process.env?.NODE;
+  if (!major && maybeNodePath) {
+    const nodeVersion = /v(\d+)\.(\d+)\.(\d+)/.exec(maybeNodePath)?.[0];
+    if (nodeVersion) {
+      [major] = nodeVersion
+        .substring(1)
+        .split(".")
+        .map((v) => parseInt(v, 10));
+    }
+  }
+
+  if (major && major < 18) {
+    throw new Error(
+      `[UT]: YOU ARE USING A LEGACY (${major}) NODE VERSION WHICH ISN'T OFFICIALLY SUPPORTED. PLEASE UPGRADE TO NODE 18+.`,
+    );
```

Comment:
```suggestion
    );
    process.exit(1);  // Kill the process if it isn't going to work correctly anyway
```

Reply:
Is that cross-runtime compatible?

Reply:
Good point. I would assume so, but don't actually know for sure. I'll have to check.

Reply:
Should work in bun. Not sure of other runtimes that have `process` defined tbh

Reply:
We can fix that when we get there if it's an issue I guess